output-validation-rule.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var assert = require('assert');
+var Joi = require('joi');
+
+module.exports = OutputValidationRule;
+
+function OutputValidationRule(status, spec) {
+  assert(status, 'OutputValidationRule: missing status param');
+  assert(spec, 'OutputValidationRule: missing spec param');
+
+  this.ranges = status.split(',').map(trim).filter(Boolean).map(rangify);
+  this.status = status;
+  this.spec = spec;
+
+  this.validateInput();
+}
+
+/**
+ * Validates it's input values
+ *
+ * @throws Error
+ */
+
+OutputValidationRule.prototype.validateInput = function validateInput() {
+  assert(/^[0-9,\-]+|\*$/.test(this.status),
+    'output validation key: "' + this.status + '" must match "0-9,-*"'
```

Comment:
"0-9,-*" looks like it means "0-9" or "-*". Might be better to spell it out the supported syntaxes, e.g. must be a comma-separated list of numbers, ranges, or *